torchgeo/datasets/geo.py

```diff
@@ -946,6 +1054,106 @@ def res(self, new_res: float) -> None:
         self.datasets[1].res = new_res
 
 
+class MultiQueryDataset(IntersectionDataset):
```

Comment:
Not sure on the naming scheme here yet.

Reply:
The current name is `MultiQueryDataset`, however, in our discussion, this is what I was referring to with `TimeSeriesDataset`

Reply:
`@` might be the best operator to use if we want to talk about "time". Maybe `+` if it's more generic and supports more than just time series. But idk what other applications it might be used for.